core/src/HPX/Kokkos_HPX.cpp

```diff
@@ -53,6 +53,9 @@ namespace Kokkos {
 namespace Experimental {
 
 bool HPX::m_hpx_initialized = false;
+bool HPX::m_was_initialized = false;
```

Comment:
Why do we need both `HPX::m_hpx_initialized` and `HPX:: m_was_initialized`?

Reply:
m_hpx_initialized is used for is_initialized(), which can be false in cases where m_was_intitialized is true.